code/tsp_ls2.py

```diff
@@ -2,12 +2,152 @@
 from tsp_types import Trace
 from tsp_types import Node
 from tsp_types import Edge
+from tsp_types import Tracepoint
 import random
 import numpy as np
 import math
 import time
+from multiprocessing import Process, Pipe, Queue
+
+def distance_calculator(nodes):
+    num_locs = len(nodes)
+    mat = np.empty((num_locs,num_locs), dtype=Edge)
+    for x in range(num_locs):
+        for y in range(num_locs):
+            mat[x,y] = Edge(nodes[x],nodes[y])
+            if x == y:
+                mat[x,y].cost = np.inf
+    return mat
+
+def get_min_cost(dest_group,rem_nodes):
+    min_cost = np.inf
+    min_index = np.inf
+    for i in range (0,dest_group.shape[0]):
+        if (dest_group[i].cost < min_cost) and rem_nodes[i] == 1:
+            min_cost = dest_group[i].cost
+            min_index = i
+    return min_cost, min_index
+
+def calculate_route_cost(route,cost_matrix):
+    cost = 0
+    for i in range(0,len(route)-1):
+        cost += cost_matrix[route[i],route[i+1]].cost
+    return cost
+
+def printmat(mat):
+    for i in range(0,len(mat)):
+        for j in range(0,len(mat)):
+            print(mat[i,j].cost, end =" ")  
+        print()   
+
+#This will solve a greedy problem to give us a place to start
+def greedy_heuristic(dist_matrix,index_start):
```

Comment:
recommend putting the greedy_heuristic function in one spot

Reply:
Removing Greedy alg from ls2 and moving it to ls1